webapp/components/emoji_picker/emoji_picker.jsx

```diff
@@ -293,9 +310,9 @@ export default class EmojiPicker extends React.Component {
 
         for (const category of CATEGORIES) {
             if (category === 'custom') {
-                items.push(this.renderCategory('custom', this.state.filter, this.props.customEmojis));
+                items.push(this.renderCategory('custom', true, this.state.filter, this.props.customEmojis));
             } else {
-                items.push(this.renderCategory(category, this.state.filter));
+                items.push(this.renderCategory(category, category === 'recent' || didPreloadCategory(category), this.state.filter));
```

Comment:
I'm trying to think of a better way to do this, but using global state in the render function is considered a bad idea in React as well as using `forceUpdate` like we're doing below. What if you
1. Replace `didPreloadCategory` with `getPreloadStatuses` which returns a copy of `preloads` (using a copy for immutability and such) which `EmojiPicker` includes in it's state
2. Instead of having the emoji picker register itself directly, have it pass in a listener that just updates the `preloads` state using `getPreloadStatuses`